### ISSUE TITLE
Record Type re-written to Jackson-incompatible Class

### ISSUE THREAD

**Issue by efowler** [Guardsquare/proguard#363]
Using Amazon Corretto JDK 20, with Spring Boot 3.0.6, I have Record type objects as @RequestBody parameters in a @RestController.  These record types are converted from raw json by Jackson.  The problem arises when ProGuard re-writes the record objects into a class object that Jackson cannot populate due to it not having a default constructor.  

So is there a setting to either disable the re-write to a class and just leave it alone as a record (preferred), or in the case of a Record, also include a default constructor?

**Comment by efowler** [comment]
The keep rules don't help, it still gets re-written to a class.  I'll work on a sample.

**Comment by pwinckles** [comment]
@mrjameshamilton A reproducible sample is any record. Even when ProGuard is configured to keep everything it rewrites it like:

```java
public final class Example extends Record {
    private final String someField;

    public Example(String var1) {
        this.someField = var1;
    }

// etc
}
```

Which completely breaks jackson.

**Comment by pwinckles** [comment]
For what it's worth, it looks like you can work around the issue by annotating the fields with `@JsonProperty(value = "fieldName")`

**Comment by abdolence** [comment]
Yep, records after Proguard aren't usable. Had to migrate to another option.

**Comment by Emmeral** [comment]
For me `-keepattributes Record` did the trick.
